abci-rs/Cargo.toml

```diff
@@ -0,0 +1,50 @@
+[package]
+name = "abci-rs"
+version = "0.10.0"
+authors = ["Devashish Dixit <devashishdxt@gmail.com>"]
+license = "MIT/Apache-2.0"
+description = "A Rust crate for creating ABCI applications"
+homepage = "https://github.com/devashishdxt/abci-rs"
+repository = "https://github.com/devashishdxt/abci-rs"
+categories = ["network-programming"]
+keywords = ["blockchain", "tendermint", "abci"]
+readme = "README.md"
+include = ["Cargo.toml", "src/**/*.rs", "README.md"]
+edition = "2018"
+
+[lib]
+name = "abci"
+
+[package.metadata.docs.rs]
+features = ["doc"]
+rustdoc-args = ["--cfg", "feature=\"doc\""]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+async-std = { version = "1.6", optional = true }
+async-trait = "0.1"
+bytes = "0.5"
+integer-encoding = { version = "1.1", optional = true }
```

Comment:
there's a small bug in the released integer-encoding: https://github.com/dermesser/integer-encoding-rs/pull/10
not sure of the impact. but it's a fairly small crate / code needed here, so could be possibly copied / vendored in here directly?

Reply:
From https://github.com/tendermint/rust-abci/pull/134 it seems we're not reading the first byte? Was it just being ignored or? Trying to understand the state of things and how this bug was surfaced in the first place

Reply:
It seems it's being read (https://github.com/tendermint/rust-abci/blob/develop/src/codec.rs#L26 -- or @yihuang why it's not?) and under normal circumstances, TM would be writing terminated varints as lengths ?

Reply:
what does terminated mean here? do we mean like a terminal byte in the varint encoding, ie. where the MSB is not set? I think that first "length of length" is always just one byte here so should always be terminal, and should probably also always be small (like 1-3, maybe max 10, since theres a limit on how big a msg tendermint would send).

Reply:
terminated varint -- the last byte has MSB set.
the code in rust-abci seems to match what's in Golang:
https://github.com/tendermint/tendermint/blob/master/abci/types/messages.go#L54